### DISCUSSION TITLE
Changing material color or  texture to a material

### DISCUSSION THREAD

**Discussion by morenostef** [stride3d/stride#2437] [Newcomers]
Hi. I'm experimenting with materials in c#. I'd like to replace a material of an imported model with a new material runtime.If the material is already created among the assets,no problem loading it . The problems arise when i create a material runtime and i must change color or texture. I followed the manual suggestion below,but there is no  "material.Parameters" and,obviously i get errors.
Are there solutions ?
/////////////////////////
    // Create a material (eg with red diffuse color).
    var materialDescription = new MaterialDescriptor
    {
        Attributes =
	    {
	        DiffuseModel = new MaterialDiffuseLambertModelFeature(),
	        Diffuse = new MaterialDiffuseMapFeature(new ComputeColor { Key = MaterialKeys.DiffuseValue })
	    }
    };
    var material = Material.New(GraphicsDevice, materialDescription);
    material.Parameters[0].Set(MaterialKeys.DiffuseValue, Color.Red);
    model.Materials.Add(0, material);
/////////////////////////////////////

**Comment by Basewq** [comment]
Can you clarify where you got this example?

The code should be
```
material.Passes[0].Parameters.Set(MaterialKeys.DiffuseValue, Color.Red);
```

**Reply by morenostef** [reply]
I found the example in the stride 4.2 manual
This is the link:
https://doc.stride3d.net/latest/en/manual/scripts/create-a-model-from-code.html

See at the end of that page.
